moped-editor/src/views/projects/projectView/ProjectComponents/BaseMapSpeedDial.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from "react";
+import clsx from "clsx";
+import { Icon, makeStyles, Typography } from "@material-ui/core";
+import { SpeedDial, SpeedDialAction } from "@material-ui/lab";
+import {
+  MAPBOX_PADDING_PIXELS,
+  MAPBOX_CONTROL_BUTTON_WIDTH,
+  COLORS,
+} from "./mapStyleSettings";
+
+const useStyles = makeStyles((theme) => ({
+  speedDialAction: {
+    width: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    height: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    backgroundSize: "100% 100%",
+    borderRadius: 4,
+    background: COLORS.mutedGray,
+  },
+  speedDialStreets: {
+    color: "black",
+    backgroundImage: `url(${process.env.PUBLIC_URL}/static/images/mapStreets.jpg)`,
+  },
+  speedDialAerial: {
+    color: "white",
+    backgroundImage: `url(${process.env.PUBLIC_URL}/static/images/mapAerial.jpg)`,
+  },
+  fabLabel: ({ isSpeedDialOpen }) => ({
+    ...(isSpeedDialOpen && { backgroundColor: "rgba(0,0,0,.25)" }),
+    height: "100%",
+  }),
+  mapStyleToggleLabel: {
+    fontSize: ".8rem",
+    fontWeight: "bold",
+  },
+  mapStyleToggleLabelIcon: {
+    position: "relative",
+    top: 3,
+    marginRight: "4px",
+  },
+  mapStyleActionLabel: {
+    position: "absolute",
+    bottom: 0,
+  },
+  speedDial: {
+    position: "absolute",
+    height: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    right: `${MAPBOX_PADDING_PIXELS}px`,
+    // Mapbox basemap has copyright info below the speedial while NearMap tiles do not
+    bottom: ({ basemapKey }) =>
+      basemapKey === "aerial"
+        ? `${MAPBOX_PADDING_PIXELS}px`
+        : `${MAPBOX_PADDING_PIXELS * 3}px`,
+    // Mapbox copyright info collapses to a taller info icon at 990px and below
+    [theme.breakpoints.down(991)]: {
+      bottom: ({ basemapKey }) =>
+        basemapKey === "aerial"
+          ? `${MAPBOX_PADDING_PIXELS}px`
+          : `${MAPBOX_PADDING_PIXELS * 4}px`,
+    },
+  },
+}));
+
+/**
+ * Generates the basemap selector using the SpeedDial component
+ * @param {function} setBasemapKey - The function we will call with to change the basemap key value
+ * @param {string} basemapKey - The name of the current basemap key
+ * @return {JSX.Element}
+ */
+const BasemapSpeedDial = ({ setBasemapKey, basemapKey }) => {
+  const [isSpeedDialOpen, setIsSpeedDialOpen] = useState(false);
+  const classes = useStyles({ basemapKey, isSpeedDialOpen });
+
+  /**
+   * Changes the current basemap and closes the speed dial menu
+   * @param basemapName
+   */
+  const onBasemapSelect = (basemapKey) => {
+    setIsSpeedDialOpen(false);
+    setBasemapKey(basemapKey);
+  };
+
+  /**
+   * Opens the speed dial menu
+   */
```

Comment:
Curious about comments like these - do we find them useful? I think the code tells the story, but I wanted to bring it up before removing them.

Reply:
i am a fan of removing redundant comments, but i dont think i docstring and comment enough so take my comment with a big grain of salt

Reply:
personally i feel like its better to over comment than under comment/not comment at all. in the end its just making the code more readable and easily accessible to new eyes, so i feel like one can never comment too much. but i also had CS professors that would have us comment almost every line of code lol

Reply:
I think Rose's instinct is the right on for when the code is not lexical or difficult to read. Things like regex patterns and CIDR IP notations and like any `sed` command I've ever written - can't comment enough.

But .. for when the code is lexical and has full, verbose variables names and function names and stuff like that, then the effort that went into making those well-named symbols is actually more valuable than repeating yourself in the comments. ← This is where you want to be. Comments are for when you can't quite get there.

>There are only two hard things in Computer Science: cache invalidation and naming things.

>-- Phil Karlton

In this case, I got what i needed from `onBaseMapSelect` and would strike the comment, and would maybe consider changing the name of the parameter in the function def to basemapName, maybe?? (This change suggestion would not normally have been even close to enough of a thing to write up a comment for, BTW.)

Reply:
Thanks, y'all! 🙏 I removed the "Opens the speed dial menu" comment but kept the other one and filled in the docstring along with fixing a typo